docs/git/cli/lesson.md

```diff
@@ -0,0 +1,87 @@
+---
+id: lesson
+title: Lesson
+sidebar_label: Lesson
+---
+
+import Feedback from "@theme/Feedback";
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Learning Objectives
+
+By the end of this class, you should be able to:
+
+- To _clone_ a repository using Git CLI
+- To _commit_ to a repository
+- To _push_ changes using Git CLI
+- To view Git's _history log_ using Git CLI
+- To view what uncommitted changes you have with git _status_
+
+---
+
+## Before you start
+
+Make sure that you've got `Terminal` / `Git Bash` / `Windows WSL` working correctly. Open a terminal window and type
+
+```
+git
+```
+
+If you get a response that starts with
+
+```
+usage: git [--version] [--help] [-C <path>] [-c <name>=<value>]....
+```
+
+then everything is working!
+
+If you see something different, make sure that git is installed correctly. You can find instructions for this [here](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+
+## 1) Why use Git on the Command Line
+
+_This section is taken from [this](https://gidgid.medium.com/why-you-should-use-git-from-the-command-line-7054150e7eb) excellent blog_
+
+Up until now you've been using Git using Github Desktop quite happily - why are we confusing you with a different way to use Git.
+
+Well, there a few really good reasons...
+
+### You feel comfortable everywhere
+
+A big advantage for using Git CLI is that you work comfortably everywhere, the experience of working on your local computer, a colleague computer or even a remote machine with a different OS feels more or less the same (minus your personal configurations perhaps).
+
+### Fully utilizing Git
+
+The big downside of using GUI for Git is that it exposes you to a relatively small interface (whatever part the GUI creator deemed important), this basically means you are not fully interacting with Git and are therefore not familiar with (at least) some of its capabilities.
+
+There are some really good developers that used a certain GUI and were not even aware that Git has a staging area. When using the Git CLI you are using the entire Git interface and are therefore able to see the bigger picture, this makes you much more comfortable with Git and will almost certainly expose you to some useful tools you might not have used otherwise.
+
+### Git CLI gives you the power of the command line
+
+We often try to reuse & build on top of existing tools and this is no different since you’re already using the command line, you are able to compose your Git flow with many useful command-line tools. From looping over each commit and running tests on it, through easily adding files using FZF to using Ag on your logs — the only limit here is your creativity.
```

Comment:
```suggestion
We often try to reuse & build on top of existing tools and this is no different since you’re already using the command line, you are able to compose your Git flow with many useful command-line tools. From looping over each commit and running tests on it, through easily adding files using [FZF](https://www.youtube.com/watch?v=QeJkAs_PEQQ) to using [Ag](http://conqueringthecommandline.com/book/ack_ag) on your logs — the only limit here is your creativity.
```

I'm not convinced these are the _best_ resources we could be linking to, but I feel like we probably should be linking to something, rather than just saying the names of these tools...